File: day-1-security/callcatalogue.py
import requests
import getpass

context = "/d4science.research-infrastructures.eu/FARM/ITINERIS_Training-Platform"
clientid = "itineris.d4science.org"
user = "your_username"
pwd = getpass.getpass("Insert password: ")
iam = "https://accounts.d4science.org/auth/realms/d4science/protocol/openid-connect/token"


#Auth related functions
def getToken():
    logindata = { "grant_type" : "password", "client_id" : clientid, "username" : user, "password" : pwd, "scope" : "d4s-context:" + context }
    loginheaders = { "Content-Type" : "application/x-www-form-urlencoded"}
    resp = requests.post(iam, data=logindata, headers=loginheaders)
    print(resp)
    jwt = resp.json()
    return jwt

def refreshToken(jwt):
    logindata = { "grant_type" : "refresh_token", "client_id" : clientid, "refresh_token" : jwt["refresh_token"]}
    loginheaders = { "Content-Type" : "application/x-www-form-urlencoded"}
    resp = requests.post(iam, data=logindata, headers=loginheaders)
    print(resp)
    jwt = resp.json()
    return jwt

jwt = getToken()
accesstoken = jwt["access_token"]

catalogue = "https://api.d4science.org/catalogue"
query = "items?limit=10&offset=1"
url = catalogue + "/" + query

resp = requests.get(url, headers={"Authorization" : "Bearer " + accesstoken})
print(resp.json())

print(refreshToken(jwt))

File: day-1-security/callcatalogue.r
library(httr)
library(jsonlite)
library(rstudioapi)

# Init
user = 'your_username'
context = '/d4science.research-infrastructures.eu/FARM/ITINERIS_Training-Platform'
auth_ep = 'https://accounts.d4science.org/auth/realms/d4science/protocol/openid-connect/token'
client_id = 'itineris.d4science.org'

getToken <- function() {
  body = list('grant_type'='password', 'scope'=paste('d4s-context',context, sep=':'), 'client_id'=client_id, 'username'=user, 'password'=pwd)
  res = POST(auth_ep, body=body, encode=c('form'), add_headers(c('Content-Type' = 'application/x-www-form-urlencoded')))
  jwt = fromJSON(content(res, 'text'))
  return (jwt)
}

# Auth code
pwd = askForPassword(prompt='Password to login')
jwt = getToken()

catalogue = "https://api.d4science.org/catalogue"
query = "items?limit=10&offset=5"
url = paste(catalogue, query, sep="/")

resp = GET(url, add_headers(c('Authorization' = paste('Bearer', jwt["access_token"], sep=' '))))
print(fromJSON(content(resp, 'text')))